src/MigrationTools.Clients.AzureDevops.Rest/Endpoints/AzureDevOpsEndpoint.cs

```diff
@@ -247,6 +247,27 @@ public async Task<DefinitionType> GetApiDefinitionAsync<DefinitionType>(string[]
             return await httpResponse.Content.ReadAsAsync<DefinitionType>();
         }
 
+        public async Task<Project> GetProject()
+        {
+            var apiPathAttribute = typeof(Project).GetCustomAttributes(typeof(ApiPathAttribute), false).OfType<ApiPathAttribute>().FirstOrDefault();
+            if (apiPathAttribute == null)
+            {
+                throw new ArgumentNullException($"On the class defintion of '{typeof(Project).Name}' is the attribute 'ApiName' misssing. Please add the 'ApiName' Attribute to your class");
+            }
+
+            var baseUrl = new UriBuilder(Options.Organisation);
+            baseUrl.AppendPathSegments("_apis", apiPathAttribute.Path,Options.Project);
+            var client = CreateHttpClientWithHeaders(baseUrl.Uri.AbsoluteUri.ToString(), "api-version=7.0");
```

Comment:
I haven't been able to get my hands on a on prem DevOps Servier instance but I'm pretty sure that this would fail on an older version such as Azure DevOps Server 2019 since API version 7 does not exist there. We have a central plce where we choose the appropriate version. Try re-using this